docs/events_handling.md

```diff
@@ -0,0 +1,103 @@
+
+# Events handling
+
+## Subscribed events
+
+Precaution needs to subscribe to a number of events
+to satisfy different use cases of our users.
+
+The events are separated into two groups `pull requests
+events` and `check suites/runs events`.
+
+### Pull requests events
+
+Official GitHub documentation: [PullRequest event](https://developer.github.com/v3/activity/events/types/#pullrequestevent)
+
+* <strong>pull_request.opened</strong>
+
+*occurs: when a new pull request is created*
+
+*triggers: start of the Precaution scan*
+
+* <strong>pull_request.reopened</strong>
+
+*occurs - when a pull request is reopened from a closed state*
+
+*triggers: start of a new Precaution scan*
```

Comment:
I feel like we need to delineate this triggers from `pull_request.opened`. "Re-start Precaution scan?"

Reply:
Or maybe we just need to be consistent in all of the triggers. All triggered events are effectively the same, right? scanning all changes in the PR

Reply:
Yes, all triggers are the same. The pull request events are just different entry point into the Precaution workflow - which is basically a scan on all files in the pull request.